hudi-utilities/src/main/java/org/apache/hudi/utilities/HoodieMetadataTableValidator.java

```diff
@@ -914,6 +950,97 @@ private void validateAllColumnStats(
     validate(metadataBasedColStats, fsBasedColStats, partitionPath, "column stats");
   }
 
+  private void validatePartitionStats(
+      HoodieMetadataValidationContext metadataTableBasedContext,
+      Set<String> baseDataFilesForCleaning,
+      List<String> allPartitions) throws Exception {
+
+    HoodieSparkEngineContext engineContext = new HoodieSparkEngineContext(jsc);
+    HoodieData<HoodieMetadataColumnStats> partitionStatsUsingColStats = getPartitionStatsUsingColStats(metadataTableBasedContext,
+        baseDataFilesForCleaning, allPartitions, engineContext);
+
+    TableSchemaResolver schemaResolver = new TableSchemaResolver(metaClient);
+    boolean enableMetadataTable = true;
+    HoodieMetadataConfig metadataConfig = HoodieMetadataConfig.newBuilder()
+        .enable(enableMetadataTable)
+        .withMetadataIndexBloomFilter(enableMetadataTable)
+        .withMetadataIndexColumnStats(enableMetadataTable)
+        .withEnableRecordIndex(enableMetadataTable)
+        .build();
+    PartitionStatsIndexSupport partitionStatsIndexSupport =
+        new PartitionStatsIndexSupport(engineContext.getSqlContext().sparkSession(), AvroConversionUtils.convertAvroSchemaToStructType(schemaResolver.getTableAvroSchema()), metadataConfig, metaClient,
+            false);
+    HoodieData<HoodieMetadataColumnStats> partitionStats =
+        partitionStatsIndexSupport.loadColumnStatsIndexRecords(JavaConverters.asScalaBufferConverter(metadataTableBasedContext.allColumnNameList).asScala().toSeq(), false);
+    JavaRDD<HoodieMetadataColumnStats> diffRDD = HoodieJavaRDD.getJavaRDD(partitionStats).subtract(HoodieJavaRDD.getJavaRDD(partitionStatsUsingColStats));
+    if (!diffRDD.isEmpty()) {
+      List<HoodieMetadataColumnStats> diff = diffRDD.collect();
+      Set<String> partitionPaths = diff.stream().map(HoodieMetadataColumnStats::getFileName).collect(Collectors.toSet());
+      StringBuilder statDiffMsg = new StringBuilder();
+      for (String partitionPath : partitionPaths) {
+        List<HoodieMetadataColumnStats> diffPartitionStatsColStats = partitionStatsUsingColStats.filter(stat -> stat.getFileName().equals(partitionPath)).collectAsList();
+        List<HoodieMetadataColumnStats> diffPartitionStats = partitionStats.filter(stat -> stat.getFileName().equals(partitionPath)).collectAsList();
+        statDiffMsg.append(String.format("Partition stats from MDT: %s from colstats: %s", Arrays.toString(diffPartitionStats.toArray()), Arrays.toString(diffPartitionStatsColStats.toArray())));
+      }
+      throw new HoodieValidationException(String.format("Partition stats validation failed diff: %s", statDiffMsg));
+    }
+  }
+
+  private HoodieData<HoodieMetadataColumnStats> getPartitionStatsUsingColStats(HoodieMetadataValidationContext metadataTableBasedContext, Set<String> baseDataFilesForCleaning,
```

Comment:
Should we rely on colstats for partition stats validation? What if colstats itself is wrong? Maybe we can compute colstats from scratch for the latest file slices. Since this is a validator we can probably afford to incur higher overhead.

Reply:
We need to rely on colstats so that log files are also accounted for.  Created https://issues.apache.org/jira/browse/HUDI-8260 to fix the validation logic in col stats.